### ISSUE TITLE
Extend rules documentation with before/after code blocks and real use cases

### ISSUE THREAD

**Issue by alexandear** [mgechev/revive#1498]
Currently, RULES_DESCRIPTIONS.md provides descriptions of rules but lacks practical examples.
This makes it harder for developers to quickly understand the impact of enabling a rule.

### Proposal

- Add short before/after code blocks to every rule in [RULES_DESCRIPTIONS.md](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md). Can be similar to "Before/After" in [go-critic](https://go-critic.com/overview.html#deferinloop).
- Where possible, include real use cases of fixing such rules in well-known Open Source codebases.
- Ensure examples are minimal but illustrative, showing both the violation and the corrected version.

This will make the documentation more practical, reduce ambiguity, and help developers adopt Revive more effectively.

### Example

Below is a proposed format for `use-any` description.

<details><summary>Details</summary>
<p>

## use-any

**Description**

Since Go 1.18, `interface{}` has an alias: `any`. This rule proposes to replace instances of `interface{}` with `any` to follow modern Go conventions introduced with generics.

**Configuration**

N/A

**Examples**

Before (violation):

```go
func PrintValue(v interface{}) {
    fmt.Println(v)
}
```

After (fixed):

```go
func PrintValue(v any) {
    fmt.Println(v)
}
```

**Real-world use cases**

- [google/go-github](https://github.com/google/go-github/pull/3584/files#diff-ed84bc778f76d11104a43a06937beca65b07fddc6ea855e64a3e219b0fb5314bR26)
- [lima-vm/lima](https://github.com/lima-vm/lima/pull/3116/files#diff-720f301bbbcb5bcec0d0faa57b642ebe058da7c331355e064af74e8c9de91dd6R91)
- [kubernetes-sigs/kueue](https://github.com/kubernetes-sigs/kueue/pull/4807/files#diff-0ed622fe6e1b7ea6f4c458aeee1a1723fd13c7886e03c10d6c77caf9865ce008R47)

</p>
</details>
